modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -432,6 +435,10 @@ const FrontendDataSet = ({
 		});
 	}, [selectedItemsValue, items, selectedItemsKey]);
 
+	useEffect(() => {
+		onItemSelected(selectedItems);
+	}, [onItemSelected, selectedItems]);
```

Comment:
Can we reuse `onSelect` here, instead of adding `onItemSelected`? It seems to me it does the same thing. It's [only implemented for List view](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/list/List.js#L96), we only needed it there up to now. It's also implemented in a way that avoids `useEffect`. It's generally a good idea to avoid it, as it results in poorer performance and readability, and can lead to bugs difficult to solve. So, even if we need to add a new prop here, can we refactor it to avoid using `useEffect`?

Reply:
Hey @markocikos, right, I've reused the already present `onSelect` which was unused, but since there was already a `useEffect` for `selectedItems` state update in the top component, I trigger the `onSelect` in there. Would it be fine?